docs: ADR-002 The Photon Token

tbruyelle opened the pull request:
First draft of the ADR to be reviewed.

I left a few comments for reviewers after the `XXX` symbol.

Also updated the `adr-template` because it was using the windows file format.

[Rendered version](https://github.com/tbruyelle/atomone/blob/tbruyelle/doc/adr-photon/docs/architecture/adr-002-photon-token.md)

giunatale reviewed (commented):
I have proposed some changes and given some initial feedback. Will come back to this with my own writeups after you give it another pass. But in general the groundwork is there. The design is simple enough as it should be.

We should also mention maybe the fact that photon will also be used for ICS payments, but that such an enforcement will be a burden of the ICS software, I think. Maybe not relevant to this module but relevant in clarifying the functionality of the photon token.

tbruyelle commented:
@giunatale Addressed your comments, except for the ICF part, I don't know yet where to put it, and since you're willing to come back with your own write-ups, I let you find the correct location to talk about this. Thanks for the review!

tbruyelle commented on docs/architecture/adr-002-photon-token.md:
```diff
@@ -0,0 +1,221 @@
+  is time to start discussion with some wallets dev regarding that, this would
+  improve the UX.
+
+> [!NOTE]
```
removed NOTE

tbruyelle commented on docs/architecture/adr-002-photon-token.md:
```diff
@@ -0,0 +1,221 @@
+
+Given this formula, when the PHOTON supply reaches the max supply of 1 billion,
+it is no longer possible to get PHOTONs, because the conversion rate will return
+the 0 value.
```
Rephrased

tbruyelle commented on docs/architecture/adr-002-photon-token.md:
```diff
@@ -0,0 +1,221 @@
+
+### `ConversionRate` query
+
+The `ConversionRate` query returns a decimal which represents the current
```
done

tbruyelle commented on docs/architecture/adr-002-photon-token.md:
```diff
@@ -0,0 +1,221 @@
+
+## Abstract
+
+This ADR proposes to introduce the PHOTON token (ticker `photon`) as the only
```
fixed

tbruyelle commented on docs/architecture/adr-002-photon-token.md:
```diff
@@ -0,0 +1,221 @@
+The ADR proposes to create a new `photon` module to host the following
+features:
+- New `ConversionRate` query
+- New `MsgBurn` message (XXX or `MsgMint`? or `MsgConvert` ? or something else?).
```
renamed

giunatale commented:
@tbruyelle I pushed some changes, I think it's mostly ready for broader review.

tbruyelle reviewed (commented):
All good @giunatale, nice updates!

giunatale commented on docs/architecture/adr-002-photon-token.md:
```diff
@@ -0,0 +1,216 @@
+the situation in which this happens is if all ATONEs in circulation are burnt.
+This is not possible if there are ATONEs staked, which is a requirement to be
+able to produce blocks. This also means in practice that this message is
+expected to never be able to fail because of insufficient mintable PHOTONs.
```
```suggestion
expected to never be able to fail because of insufficient mintable PHOTONs.

There is however still a need to check that after calculations the amount of 
photons to mint is non-zero, which due to rounding is a possibility when 
burning very small fractions of the staking token (ATONE) supply.
```
